ethcore/src/db.rs

```diff
@@ -14,26 +14,59 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Extras db utils.
+//! Database-related utilities and constants.
 
 use std::ops::Deref;
 use std::hash::Hash;
 use std::collections::HashMap;
+use std::sync::Arc;
+
 use util::{DBTransaction, Database, RwLock};
 use util::rlp::{encode, Encodable, decode, Decodable};
+use util::journaldb::{self, Algorithm, JournalDB, Archive};
+
+// database columns
+/// Column for State
+pub const COL_STATE: Option<u32> = Some(0);
+/// Column for Block headers
+pub const COL_HEADERS: Option<u32> = Some(1);
+/// Column for Block bodies
+pub const COL_BODIES: Option<u32> = Some(2);
+/// Column for Extras
+pub const COL_EXTRA: Option<u32> = Some(3);
+/// Column for Traces
+pub const COL_TRACE: Option<u32> = Some(4);
+/// Column for archived state.
+pub const COL_STATE_ARCHIVE: Option<u32> = Some(5);
+/// Number of columns in DB
+pub const NUM_COLUMNS: Option<u32> = Some(6);
+
+/// Create a new `JournalDB` trait object.
+pub fn make_journaldb(backing: Arc<Database>, algorithm: Algorithm) -> Box<JournalDB> {
```

Comment:
`make_` as well as `get_` should be probably avoided for code without (logical) side effects